cli/flox/src/commands/services/mod.rs

```diff
@@ -103,6 +110,58 @@ fn processes_by_name_or_default_to_all<'a>(
     }
 }
 
+/// Note that this must be called within an existing activation, otherwise it
+/// will leave behind a process-compose since it doesn't start a watchdog.
+pub async fn start_with_new_process_compose(
+    config: Config,
+    flox: Flox,
+    environment_select: EnvironmentSelect,
+    mut concrete_environment: ConcreteEnvironment,
+    names: &[String],
+) -> Result<Vec<String>> {
+    let environment = concrete_environment.dyn_environment_ref_mut();
+    let lockfile = environment.lockfile(&flox)?;
+    let LockedManifest::Catalog(lockfile) = lockfile else {
+        unreachable!("at least it should be after https://github.com/flox/flox/issues/1858")
+    };
+    for name in names {
+        if !lockfile.manifest.services.contains_key(name) {
+            return Err(service_does_not_exist_error(name));
+        }
+    }
```

Comment:
question: does this mean to catch services that got removed?
maybe worth adding a short comment since on ffirst sight it looks like a duplication of code we already have.

Reply:
Yeah, it's covered by this test for `restart`: https://github.com/flox/flox/pull/1923/files#diff-86362a3210f85a17e9cd2c0976165a63a67fc5c9930a71d0c68d380fd98a0447R434-R450

We can't use `processes_by_name_or_default_to_all` because we don't have a running `process-compose` at this stage so we have to check against the new names from the locked manifest. I can add a comment too 👍🏻